### PR TITLE
feat: atom and list done

### PR DESCRIPTION
**Pull request by zypeh**

### CODE REVIEW

File: src/node.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#[derive(Clone, PartialEq, Debug)]
+pub enum Node {
+    Atom(Atom),
+    List(List),
+}
+
+pub type Atom = &'static str;
```
Comment (hch12907) [inline src/node.rs:7]: Does `Atom` actually use static strings? Sounds like a job for String to me

File: src/node.rs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+// =====================================
+// List
+// =====================================
+pub type List = Vec<Node>;
```
Comment (hch12907) [inline src/node.rs:12]: Use newtype pattern here
```
struct List(Vec<Node>);
```

File: src/node.rs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+pub fn cons(list: List, atom: Atom) -> Node {
```
Comment (hch12907) [inline src/node.rs:14]: With newtype, use `impl List { ... }` here

Comment (hch12907) [reply src/node.rs:14]: Ditto for `car` and `cdr`

File: src/node.rs (hunk 1 above)
Comment (zypeh) [reply src/node.rs:7]: Because we are directly writing programs on the test case. So I didn't think much and choose static string for that.

**Pull request merged**